### PR TITLE
 (maint) Update Gemfile and dependencies

### PR DESCRIPTION
**Pull request by smcclellan**
This includes an update to the new `expect` rspec syntax.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,42 +1,55 @@
...
-    ruby-progressbar (1.1.1)
-    simplecov (0.7.1)
-      multi_json (~> 1.0)
-      simplecov-html (~> 0.7.1)
-    simplecov-html (0.7.1)
-    thor (0.18.1)
+    CFPropertyList (3.0.0)
+    diff-lcs (1.3)
+    docile (1.3.2)
+    facter (2.5.1)
```
Comment (smcclellan) [inline Gemfile.lock]: What changes can we expect when going from facter 1.7.2 to 2.5.1? Anything not backwards compatible?

Comment (branan) [reply Gemfile.lock]: Facter 2 introduced structured facts, which might require some manual investigation into how the Razor server handles deserializing and storing them. API-wise it also has some changes to support those structured facts - I don't know what we support as far as custom facts in the MK

Comment (smcclellan) [reply Gemfile.lock]: That is what I suspected. Are structured facts additions only, or is it breaking for legacy facts?

Comment (branan) [reply Gemfile.lock]: All legacy facts will still exist, but I don't think there's an API to get *only* the legacy facts, so if anything downstream can't handle the structure we'll need to filter it appropriately

**Pull request merged**